### ISSUE TITLE
Get a list of ALL Directory Security Roles and make it reference-able

### ISSUE THREAD

**Issue by DevopsMercenary** [hashicorp/terraform-provider-azuread#1086]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritise this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritise the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

azuread_directory_roles only returns ACTIVE roles.  Would it be possible to add a flag to return ALL roles. Granted I can do this
```
resource "azuread_directory_role" "security-reader" {
  display_name = "Security reader"
}

resource "azuread_directory_role_assignment" "information-security-read-only" {

  role_id = azuread_directory_role.security-reader.template_id

  principal_object_id = azuread_group.information-security-read-only.object_id

}
```

but what I really want is something like `data.azuread_application_published_app_ids` where I can get ALL of the IDs for directory roles and then reference them like  `data.azuread_directory_role_ids.well_known.result[ "Security reader"]`
 I'm pretty sure that syntax is wrong but I hope it communicates the idea.

**Comment by manicminer** [comment]
Hi @DevopsMercenary, thanks for suggesting this feature. Unfortunately due to API constraints, we won't be able to implement this currently. Directory roles are only returned by the API once they have been activated in a tenant (or if they are activated by default for new tenants). If you need a specific directory role to show up in this list, you will indeed have to configure an `azuread_directory_role` resource for it.

Although we can't implement this right now due to API constraints, I'm happy to heave this issue open as a feature request and we'll revisit this in future if this constraint is lifted.

**Comment by manicminer** [comment]
Thanks to @MattGarnerAWR I believe this is now resolved with #1152
